packages/lit-dev-content/site/docs/frameworks/react.md

```diff
@@ -0,0 +1,176 @@
+
+React can already render web components, since custom elements are just HTML elements and React knows how to render HTML. But React makes some assumptions about HTML elements that don't always hold for custom elements, and it treats lower-case tag names differently from upper-case component names in ways that can make custom elements harder than necessary to use.
+
+For instance, React assumes that all JSX properties map to HTML element attributes, and provides no way to set properties. This makes it difficult to pass complex data (like objects, arrays, or functions) to web components. React also assumes that all DOM events have corresponding "event properties" (`onclick`, `onmousemove`, etc), and uses those instead of calling `addEventListener()`. This means that to properly use more complex web components you often have to use `ref()` and imperative code. (For more information on the limitations of React's web component integration, see [Custom Elements Everywhere](https://custom-elements-everywhere.com/libraries/react/results/results.html).)
+
+React is working on fixes to these issues, but in the meantime, our wrappers take care of setting properties and listening to events for you.
+
+The `@lit-labs/react` package provides two main exports:
+
+-   `createComponent()` creates a React component that _wraps_ an existing web component. The wrapper allows you to set props on the component and add event listeners to the component like you would any other React component.
+
+-   `useController()` lets you use a Lit reactive controller as a React hook.  
+
+## createComponent
+
+The `createComponent()` function makes a React component wrapper for a custom element class. The wrapper correctly passes React `props` to properties accepted by the custom element and listens for events dispatched by the custom element.
+
+### Usage
+
+Import `React`, a custom element class, and `createComponent`.
+
+```js
+import React from 'react';
+import {createComponent} from '@lit-labs/react';
+import {MyElement} from './my-element.js';
+
+export const MyElementComponent = createComponent({
+  tagName: 'my-element',
+  elementClass: MyElement,
+  react: React,
+  events: {
+    onActivate: 'activate',
+    onChange: 'change',
+  },
+});
+```
+
+After defining the React component, you can use it just as you would any other React component.
+
+```jsx
+<MyElementComponent
+  active={isActive}
+  onActivate={(e) => setIsActive(e.active)}
+  onChange={handleChange}
+/>
+```
+
+#### Options
+
+`createComponent` takes an options object with the following properties:
+
+- `tagName`: The custom element's tag name.
+- `elementClass`: The custom element class.
+- `react`: The imported `React` object. This is used to create the wrapper component with the user supplied `React`. This can also be an import of `preact-compat`.
+- `events`: An object that maps an event handler prop to an event name fired by the custom element.
+
+#### Typing event handler props
```

Comment:
I think we should add a section on slots, just pointing out that React children render to the default slot, and if you want to render to a different slot you need to use the standard `slot` attribute, and that means if you want to slot a React component to a named slot you need to wrap it in an element like `<div slot="foo">` and that you can use `display: contents` to hide that element...

Reply:
Good idea. Added the section above the typing event one as it seems more relevant/important. And linked to the slot example in the playground too.

Reply:
I do think that this could still be just "Events" with an initial paragraph describing the event prop to event name mapping in more detail. Maybe saying what naming conventions are (while the name can be whatever you want, they should start with `on`, etc), and that you need to make sure that the event name doesn't collide with other properties on the element.

Reply:
Regarding prop naming convention, future React support looks like it'll straight up take anything after `on` as the event name. 

Implementation: https://github.com/facebook/react/blob/161f6ae42cf570d3daec355abcf3b006c58315da/packages/react-dom-bindings/src/client/DOMPropertyOperations.js#L163-L171

Tests:
https://github.com/facebook/react/blob/161f6ae42cf570d3daec355abcf3b006c58315da/packages/react-dom/src/__tests__/DOMPropertyOperations-test.js#L173-L213

I'm thinking our example follow this and do `onmy-event` for the prop name that gets called for `my-event`.

Reply:
yeah, that's a great idea. Mention that re recommendation is to be compatible with React on this front.